src/main/java/PalindromeChecker.java

```diff
@@ -0,0 +1,14 @@
+public class PalindromeChecker {
+
+    String isPalindrome(String str) {
```

Comment:
trochę za dużo się tu dzieje jak na jedną metodę, która nazywa się "isPalindrome".


zazwyczaj "is" w anzwie metody zwraca boolean - isEven/isCorrect/isSuccess/isPalindrome.
W twoim przypadku "is" zwraca string.

Co więcej - dla pytania: "is palindrome?" czasem odpowiadasz "Text length is..."/ "There is no text to check".

Pomyśl tutaj o jakimś ułożeniu tego. 
isPalindrome to tylko to:
```
      for (int i = 0; i < n / 2; i++) {
                if (str.charAt(i) != str.charAt(n - i - 1)) return false
}
 return true
```

i ta metoda powinna zwracać boolean. 

Wystarczy jak tę metodę wywołasz w innej metodzie w której sprawdzisz ten warunek numer 1 & 3.